args_test.go

```diff
@@ -273,10 +274,21 @@ func testGetDockerConfigsNoError(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, testContainerImageName, args.ContainerImageName)
 	assert.Equal(t, testContainerImageID, args.ContainerImageID)
-	assert.Equal(t, true, reflect.DeepEqual(args.ContainerLabels, testContainerLabelsMap))
-	for i := range args.ContainerEnv {
-		assert.Equal(t, testContainerEnvSlice[i], args.ContainerEnv[i])
+
+	// Docker logging use a slice to store the environment variables
+	// Each item is in the format of "key=value" the unit test should
+	// convert it back to a map to guarantee content exists but disregard
+	// the order.
+	// ref: https://github.com/moby/moby/blob/c833222d54c00d64a0fc44c561a5973ecd414053/daemon/logger/loginfo.go#L60
+	testContainerEnvMap := make(map[string]struct{})
+	for _, v := range testContainerEnvSlice {
+		testContainerEnvMap[v] = struct{}{}
+	}
+	argsContainerEnvMap := make(map[string]struct{})
+	for _, v := range args.ContainerEnv {
+		argsContainerEnvMap[v] = struct{}{}
 	}
+	assert.Equal(t, true, reflect.DeepEqual(testContainerEnvMap, argsContainerEnvMap))
```

Comment:
The previous test case is to check if `ContainerEnv` is a subset of `testContainerEnvSlice` and the newer one is to check if both are same, which is more strict. I don't know much about these but I think it's okay.

Reply:
I don't think previously it was intended to check if it's a subset  - only that it's implementation suggests it. 

Given the inputs that is being used for the test case, I think it's actually intended to check if the values are the same:

```
	testContainerEnv       = "{\"env0\":\"envValue0\",\"env1\":\"envValue1\"}"
)
var (
	testContainerLabelsMap = map[string]string{
		"label0": "labelValue0",
		"label1": "labelValue1",
	}
	testContainerEnvSlice = [2]string{
		"env0=envValue0",
		"env1=envValue1",
	}
)
```